### ISSUE TITLE
Remember Signing Settings (Regression?)

### ISSUE THREAD

**Issue by RickStrahl** [NuGetPackageExplorer/NuGetPackageExplorer#985]
**Type** (choose one):

- Bug

NuGet Package Explorer - Chocolatey - X64 - (5.6.70+e3d04185cb)

Win10 2004 64bit

**Installed from**:  Zip on GitHub

In case of a BUG:

It looks like the signing dialog no longer remembers a previously selected certificate like a it used to (or maybe if it's the only available signing cert?) nor the timestamp server.

This used to work until I recently upgraded to the non-store version. It looks like the app does remember the Nuget publish key so it is still storing some information.

**Comment by RickStrahl** [comment]
@campersau - you mentioned you tried this and didn't get this to fail which is odd since this fails for me consistently?

I've since uninstalled and reinstalled again the latest version 5.7.1 from Chocolatey and still see the same behavior. 

The certificate isn't a big deal really and I could understand if that doesn't preserve, since I can easily pick it again and there's only one signing cert installed.  But remembering that damn timestamp server is a pain as I have to go look it up in some other signing code to remember each time :-).

**Comment by campersau** [comment]
Yes, the settings were stored and restored correctly. I have also used the 5.7.1 version from Chocolatey.

Can you please check the `user.config` setting file in a subfolder of  `%userprofile%\appdata\local\Luan_Nguyen,Julian_Verdur`?
Maybe you can try deleting them and see if it works afterwards?

**Comment by RickStrahl** [comment]
So gave that a try but no luck:

* Removed folders in that hierarchy
* Signed a package
* Recompiled
* Go back to sign it again and the Package info is blank again

![image](https://user-images.githubusercontent.com/1374013/82610172-9efd0c80-9b59-11ea-8600-d4f24438b88e.png)

The config recreates in the folder, but after signing I see no info related to the cert in there:

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <userSettings>
        <PackageExplorer.Properties.Settings>
            <setting name="WindowPlacement" serializeAs="String">
                <value>44|0|1|-1;-1|-1;-1|364;364;1114;1014</value>
            </setting>
            <setting name="IsFirstTime" serializeAs="String">
                <value>False</value>
            </setting>
            <setting name="MruFiles" serializeAs="Xml">
                <value>
                    <ArrayOfString xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                        xmlns:xsd="http://www.w3.org/2001/XMLSchema">
                        <string>Westwind.Utilities|3.0.38|C:\projects\Westwind.Utilities\Westwind.Utilities\bin\Release\Westwind.Utilities.3.0.38.nupkg|LocalPackage</string>
                    </ArrayOfString>
                </value>
            </setting>
        </PackageExplorer.Properties.Settings>
    </userSettings>
</configuration>
```

**Comment by RickStrahl** [comment]
I should also mention that after the package is signed Package Explorer **just goes away** - it kind of looks like it's crashing silently but there's no error message or notification that it worked or not. The package is signed, but PE is gone and I have to restart to publish...

**Comment by campersau** [comment]
Did you select the certificate from "Browser..." (e.g. filesystem) or did you use the "Certificate Store..."?
If it is the latter then this if fixed by #987 which by your description looks likely. 

> I should also mention that after the package is signed Package Explorer **just goes away** - it kind of looks like it's crashing silently but there's no error message or notification that it worked or not. The package is signed, but PE is gone and I have to restart to publish...

When the application is crashing no settings will be saved to disk. But I didn't thought about that when I first debugged this issue because it was clear to me when I saw the unhandled exception crash and assumend that this would be reported as an "application crash" rather than "settings not beeing saved" but for an enduser this isn't obviouse without an error message... sorry.

**Comment by clairernovotny** [comment]
I just released 5.1.27, which has the #987 fix.

**Comment by RickStrahl** [comment]
Thanks to both of you...  that looks like it takes care of

* Signing now doesn't just 'disappear' PE
* Settings are preserved

So it looks like the issue was indeed that PE was crashing, but it would be nice if there was some sort of notification that let you know if that happens. Just disappearing I kind of figured it was because it couldn't reopen the already cached package...

IAC, thanks again. Fixed.
